libs/core-ui/src/lib/util/ObjectDetectionStatisticsUtils.ts

```diff
@@ -15,13 +15,29 @@ export enum ObjectDetectionMetrics {
 }
 
 export const generateObjectDetectionStats: (
-  selectionIndexes: number[][]
+  selectionIndexes: number[][],
+  objectDetectionCache: Map<string, [number, number, number]>,
+  objectDetectionInputs: [string, string, number]
 ) => ILabeledStatistic[][] = (
-  selectionIndexes: number[][]
+  selectionIndexes: number[][],
+  objectDetectionCache: Map<string, [number, number, number]>, //  = new Map()
+  objectDetectionInputs: [string, string, number]
 ): ILabeledStatistic[][] => {
   return selectionIndexes.map((selectionArray) => {
     const count = selectionArray.length;
 
+    const key: [number[], string, string, number] = [
+      selectionArray,
+      objectDetectionInputs[0],
+      objectDetectionInputs[1],
+      objectDetectionInputs[2]
+    ];
+    const value = objectDetectionCache.get(key.toString());
+    console.log('retrieving key');
+    console.log(key.toString());
```

Comment:
should the console.log calls be removed here?

Reply:
Yes I put in a separate comment for myself above :D